snippets/csharp/System.Collections.Generic/IEqualityComparerT/Overview/program.cs

```diff
@@ -2,81 +2,69 @@
 using System;
 using System.Collections.Generic;
 
-class Example
+static class Example
 {
-   static void Main()
-   {
-      BoxEqualityComparer boxEqC = new BoxEqualityComparer();
-
-      var boxes = new Dictionary<Box, string>(boxEqC);
-
-      var redBox = new Box(4, 3, 4);
-      AddBox(boxes, redBox, "red");
+    static void Main()
+    {
+        BoxEqualityComparer comparer = new();
 
-      var blueBox = new Box(4, 3, 4);
-      AddBox(boxes, blueBox, "blue");
+        Dictionary<Box, string> boxes = new(comparer);
 
-      var greenBox = new Box(3, 4, 3);
-      AddBox(boxes, greenBox, "green");
-      Console.WriteLine();
+        AddBox(new Box(4, 3, 4), "red");
+        AddBox(new Box(4, 3, 4), "blue");
+        AddBox(new Box(3, 4, 3), "green");
 
-      Console.WriteLine("The dictionary contains {0} Box objects.",
-                        boxes.Count);
-   }
+        Console.WriteLine($"The dictionary contains {boxes.Count} Box objects.");
 
-   private static void AddBox(Dictionary<Box, String> dict, Box box, String name)
-   {
-      try {
-         dict.Add(box, name);
-      }
-      catch (ArgumentException e) {
-         Console.WriteLine("Unable to add {0}: {1}", box, e.Message);
-      }
-   }
+        void AddBox(Box box, string name)
+        {
+            try
+            {
+                boxes.Add(box, name);
+            }
+            catch (ArgumentException e)
+            {
+                Console.WriteLine($"Unable to add {box}: {e.Message}");
+            }
+        }
+    }
 }
 
-public class Box
+class Box
 {
-    public Box(int h,  int l, int w)
-    {
-        this.Height = h;
-        this.Length = l;
-        this.Width = w;
-    }
-
-    public int Height { get; set; }
-    public int Length { get; set; }
-    public int Width { get; set; }
+    public int Height { get; }
+    public int Length { get; }
+    public int Width { get; }
 
-    public override String ToString()
+    public Box(int height, int length, int width)
     {
-       return String.Format("({0}, {1}, {2})", Height, Length, Width);
+        Height = height;
+        Length = length;
+        Width = width;
     }
+
+    public override string ToString() => $"({Height}, {Length}, {Width})";
 }
 
 class BoxEqualityComparer : IEqualityComparer<Box>
 {
-    public bool Equals(Box b1, Box b2)
+    public bool Equals(Box? b1, Box? b2)
```

Comment:
Most of this boilerplate becomes unnecessary if users convert `Box` into a record, since they get the same IEquatable semantics for free by the compiler. Does the linking documentation make reference of this?

Reply:
Agree. This is called out above, but if we make it a record then there's no sense in having a custom comparer as-is. Perhaps we need an example for which a custom comparer makes more sense, at which point we can make this a record and have the focus of the documentation be directed more towards the comparer itself.

Reply:
I'll merge this PR with this sample. 

One possibility is to define a small scenario where some, but not all, data properties participate in the equality comparison. My first idea: an IoT sensor that returns an object the has multiple measurements and a timestamp. Two measurements are considered "equal" if all the measurements are the same. The timestamp isn't part of the "equality".

That might be larger than we want in an API sample.